homeassistant/components/tag/__init__.py

```diff
@@ -119,7 +190,7 @@ async def async_scan_tag(
     if DOMAIN not in hass.config.components:
         raise HomeAssistantError("tag component has not been set up.")
 
-    helper = hass.data[DOMAIN][TAGS]
+    helper: TagStorageCollection = hass.data[DOMAIN][TAGS]
```

Comment:
We should not add more according to this pattern, instead we should use `HassKey`. This can be done in a separate PR which does other unrelated cleanup